cli/pack/common.go

```diff
@@ -458,9 +458,14 @@ func prepareDefaultPackagePaths(opts *config.CliOpts, packagePath string) {
 func getVersion(packCtx *PackCtx, opts *config.CliOpts, defaultVersion string) string {
 	packageVersion := defaultVersion
 	if packCtx.Version == "" {
-		// Get version from git only if packing an application from the current directory.
-		if opts.App.InstancesEnabled == "." {
-			version, err := util.CheckVersionFromGit(opts.App.InstancesEnabled)
+		// Get version from git only if packing an application from the current directory,
+		// or packing with cartridge-compat enabled.
+		var appPath = opts.App.InstancesEnabled
+		if packCtx.CartridgeCompat {
+			appPath = filepath.Join(appPath, packCtx.Name)
+		}
+		if opts.App.InstancesEnabled == "." || packCtx.CartridgeCompat {
+			version, err := util.CheckVersionFromGit(appPath)
 			if err == nil || version != "" {
```

Comment:
Should't it be && instead of ||?

Reply:
We still need to get version even without `cartridge-compat` flag enabled. In this case we can only get correct version if and only if `InstancesEnabled` is set to `.` (in case we might be packing several applications).

But if `cartridge-compat` is enabled, even if `InstancesEnabled` is not a `.`, we are sure that we are packing only one application (it was checked previously).

Reply:
Seems like it was left aster some rewriting of the `CheckVersionFromGit` function. Now, `err != nil` means `version == ""`. so we can change the condition to just `if err == nil` I think.